infrastructure/builder/src/dev/helm.js

```diff
@@ -1,56 +1,148 @@
-    throw new Error('Must have configured dev-config.yml to run helm.');
-  }
-
-  const namespace = config.meta.deploymentPrefix;
+const resourceTypes = [
+  'cronjob',
+  'deployment',
+  'ingress',
+  'rolebinding',
+  'role',
+  'secret',
+  'serviceaccount',
+  'service',
+];
 
-  if (action !== 'template') {
-    let res = child_process.spawnSync('kubectl', ['create', 'namespace', namespace]);
-    let output = res.stdout.toString() + res.stderr.toString();
-    if (res.status !== 0) {
-      if (!output.includes('AlreadyExists')) {
-        throw new Error(output);
+const actions = [
+  {
+    title: 'Check dev-config.yml',
+    requires: [],
+    provides: ['dev-config'],
+    run: async (requirements, utils) => {
+      const config = await readRepoYAML('dev-config.yml');
+      if (!config.meta || !config.meta.deploymentPrefix) {
+        throw new Error('Must have configured dev-config.yml to deploy.');
       }
-    }
-
-    res = child_process.spawnSync('kubectl', ['config', 'set-context', '--current', '--namespace', namespace]);
-    output = res.stdout.toString() + res.stderr.toString();
-    console.log(output);
-    if (res.status !== 0) {
-      throw new Error('Failed to update kubectl context');
-    }
-  }
-
-  let extraArgs = [];
-  if (action !== 'uninstall') {
-    extraArgs = extraArgs.concat([
-      '-f',
-      './dev-config.yml',
-      './infrastructure/k8s',
-    ]);
-  }
-
-  const helm = child_process.spawn('helm', [
-    action,
-    'taskcluster',
-    ...extraArgs,
-  ], {
-    cwd: REPO_ROOT,
-  });
-
-  helm.stdout.on('data', d => console.log(d.toString()));
-  helm.stderr.on('data', d => console.log(d.toString()));
-  return new Promise((resolve, reject) => {
-    helm.on('exit', (code, signal) => {
-      if (code !== 0) {
-        return reject('helm command failed');
+      return {'dev-config': config};
+    },
+  },
+  {
+    title: 'Helm version detection',
+    requires: [],
+    provides: ['helm-version'],
+    run: async (requirements, utils) => {
+      const res = await execCommand({
+        command: ['helm', 'version'],
+        dir: REPO_ROOT,
+        keepAllOutput: true,
+        ignoreReturn: true, // Helm 2 yells about tiller
+        utils,
+      });
+      const match = /SemVer:"(v[0-9]+.[0-9]+.[0-9]+)"/g.exec(res);
+      if (!match) {
+        throw new Error(`Could not determine helm version from: ${res}`);
+      } else if (match[1].includes('v3')) {
+        return {'helm-version': 3};
+      } else if (match[1].includes('v2')) {
+        return {'helm-version': 2};
+      } else {
+        throw new Error(`Must use supported helm version (2 or 3). You have ${match[1]}`);
+      }
+    },
+  },
+  {
+    title: 'Generate k8s Resources',
+    requires: ['dev-config', 'helm-version'],
+    provides: ['target-templates'],
+    run: async (requirements, utils) => {
+      let command = ['helm', 'template'];
+      if (requirements['helm-version'] === 3) {
```

Comment:
does this allow configuring to pick a helm version? if so, could you drop a note about it in the docs somewhere?

Reply:
Doesn't allow configuring, it just works with whatever helm you have installed (2 or 3, who knows about 1). I'll mention in the docs though how this works